src/wp-includes/option.php

```diff
@@ -1133,6 +1142,94 @@ function delete_option( $option ) {
 	return false;
 }
 
+/**
+ *  Determines the appropriate autoload value for an option based on input.
+ *
+ *  This function checks the provided autoload value and returns a standardized
+ *  value ('yes', 'no', 'auto-yes', or 'auto-no') based on specific conditions.
+ *
+ * If no valid autoload value is provided, the function will check the size of the given option value
+ * and compare it to the maximum allowed size. If it ls larger, it will return 'auto-no', otherwise
+ * it will return 'auto-yes'.
+ *
+ * @since 6.5.0
+ * @access private
+ *
+ * @param string           $option   The name of the option.
+ * @param mixed            $value    The value of the option to check its autoload value.
+ * @param string|bool|null $autoload The autoload value to check.
+ *                                   Accepts 'yes'|true to enable or 'no'|false to disable, or
+ *                                   'auto-yes' and 'auto-no' for internal purposes.
+ *                                   Any other autoload value will be forced to either `auto-yes`
+ *                                   or `auto-no`.
+ * @return string Returns the original $autoload value if valid, or 'auto-no' if the size exceeds
+ *                the maximum allowed size, or 'auto-yes' otherwise.
+ */
+function determine_option_autoload_value( $option, $value, $autoload ) {
```

Comment:
Better to use `_determine_option_autoload_value` before private function name. I got similar feedback on one of my PR but couldn't find it at the moment. Also could we use `wp` prefix? i.e. `_wp_determine_option_autoload_value`

Reply:
I'd prefer to go with `wp_determine_option_autoload_value()`. Discussions on whether or not to prefix with an underscore come up all the time, but many years ago a few long-standing core committers told me that this is a bad pattern, so I think it's best to avoid it and simply rely on `@access private`.